kasa/smart/smartdevice.py

```diff
@@ -357,12 +357,25 @@ def alias(self) -> str | None:
     @property
     def time(self) -> datetime:
         """Return the time."""
-        if self._parent and Module.Time in self._parent.modules:
-            _timemod = self._parent.modules[Module.Time]
-        else:
-            _timemod = self.modules[Module.Time]
+        if (self._parent and (time_mod := self._parent.modules.get(Module.Time))) or (
+            time_mod := self.modules.get(Module.Time)
+        ):
+            return time_mod.time
 
-        return _timemod.time
+        # We have no device time, use current local time.
+        return datetime.now(timezone.utc).astimezone().replace(microsecond=0)
+
+    @property
```

Comment:
Not that important, but perhaps keep this in its original place to keep the diff smaller?

Reply:
I did think that but in the end I thought it better to have time and on_since next to each other and as it was a fairly small diff...